### PR TITLE
fix(rate-limit): honor zero token refill rate

### PR DESCRIPTION
**Pull request by Moersity**
### Problem

`rate_limit_tokens_per_second = 0` passes configuration validation and is supported by `TokenBucket` as a zero-refill concurrency limiter. However, startup filtered out zero and fell back to `max_concurrent_requests` as the refill rate. The Helm chart also omitted the CLI flag when the value was zero.

### Solution

Preserve explicit non-negative refill rates when constructing the token bucket, and render the Helm CLI argument for zero. Keep negative values filtered so unchecked configurations cannot be cast to an invalid `usize`.

## Changes

- Honor an explicit zero token refill rate in `AppContextBuilder`.
- Pass `auth.rateLimitTokensPerSecond: 0` through the Helm chart.
- Add a regression test proving that zero disables time-based refill while returned tokens remain reusable.

## Test Plan

- `cargo +nightly fmt --all -- --check`
- `cargo clippy --all-targets --all-features -- -D warnings`
- `cargo test`
- `helm lint deploy/helm/smg`
- Render the chart with `auth.rateLimitTokensPerSecond=0` and verify that `--rate-limit-tokens-per-second 0` is present.
- Render the chart with the default `-1` and verify that the rate-limit argument is absent.

<details>
<summary>Checklist</summary>

- [x] `cargo +nightly fmt` passes
- [x] `cargo clippy --all-targets --all-features -- -D warnings` passes
- [ ] (Optional) Documentation updated
- [ ] (Optional) Please join us on Slack [#sig-smg](https://slack.lightseek.org) to discuss, review, and merge PRs

</details>

### CODE REVIEW

File: model_gateway/src/app_context.rs [hunk 1]
```diff
@@ -715,6 +715,26 @@ mod tests {
...
+            .rate_limiter
+            .expect("rate limiter should be enabled");
+
+        assert!(bucket.try_acquire(2.0).is_ok());
+        tokio::time::sleep(Duration::from_millis(20)).await;
+        assert!(bucket.try_acquire(1.0).is_err());
+
+        bucket.return_tokens_sync(1.0);
+        assert!(bucket.try_acquire(1.0).is_ok());
+    }
```
[the remark this answers is not in the document]
Comment (Moersity) [reply model_gateway/src/app_context.rs:738]: Fixed in fda958b8 — capacity 10 with a 150 ms wait replenishes about 1.5 tokens under the old fallback; I verified the test fails with the old condition and passes with zero refill.
